.github/scripts/govulncheck-submit-issues.sh

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env bash
+# SPDX-License-Identifier: MPL-2.0
+
+# This requires a "results" directory containing files with the name of the branch that the results are associated with.
+# The files inside need to contain just a one liner JSON array like: ["vulnKey1","vulnKey2",...].
+# Eg: `$ tail results/*`
+#  ==> results/main <==
+#  []
+#
+#  ==> results/v1.7 <==
+#  ["GO-2024-2947","GO-2024-2948","GO-2025-3447","GO-2025-3487","GO-2025-3503"]
+#
+#  ==> results/v1.8 <==
+#  ["GO-2025-3447","GO-2025-3487","GO-2025-3503"]
+#
+#  ==> results/v1.9 <==
+#  ["GO-2025-3447","GO-2025-3487","GO-2025-3503"]
+
+github_run_url="${1}"
+
+# This first part converts the results of type ["vulnKey1","vulnKey2",...] from the files inside the "results" directory
+# to {"vulnKey1": ["main", "v1.7", ...], "vulnKey2": ["main", "v1.7", ...]}
+
+# Start the script with an empty json object
+vuln_to_versions="{}"
+cd results
+for version in *; do
+  while IFS= read -r vuln;
+  do
+    [[ -z "${vuln}" ]] && continue
+    # Actions from this like:
+    # * is giving the shell var for the vulnerability key to jq (--arg vl "${vuln}")
+    # * is giving the shell var that contains the scanned version to jq (--arg vers "${version}"
+    # * is getting from vuln_to_versions the key name that needs to be a vulnerability key and assigns to its array the currently proceesed version ('.[$vl] = .[$vl] + [$vers]')
+    # * this is generating outputs like {"vulnKey": ["version1", "version2"]}. Eg: {"GO-2024-2947":["v1.7"]}
+    # * in the end overwrites the content of vuln_to_versions with the newly generated content
+    vuln_to_versions="$(echo "${vuln_to_versions}" | jq -c --arg vl "${vuln}" --arg vers "${version}" '.[$vl] = .[$vl] + [$vers]')"
+  done <<< "$(cat $version | jq -r '.[]')" # This one is exploding a json array into multiple lines
+done
+
+# This second part is just using the ".key" that is the vulnerability key and ".value" that is the list of affected version(s)
+# to generate the commands to create GitHub issues.
+while IFS= read -r vuln;
+do
+  vuln_key="$(echo ${vuln} | jq -r '.key')"
+  affected_versions="$(echo ${vuln} | jq -r '.value[]' | xargs)"
+  ticket_title="${vuln_key} reported"
+
+  reported_issues="$(gh issue -R opentofu/opentofu list --search "\"${ticket_title}\" state:open" --json number)"
```

Comment:
I assume that the `state:open` here means that `reported_issues` would only recognize issues that are currently open.

In the case of a false positive we would effectively leave an issue "unresolved" on a branch and justify in our advisory why the problem doesn't actually affect OpenTofu in practice, so perhaps we should also allow this to include issues that are already closed so that in the false positive case we can close the issue and this script will skip creating a duplicate?

(I expect that `govulncheck`'s reporting will be precise enough that false positives won't arise _often_, but I'd like to make sure we have a process to deal with them if they _do_ arise, and letting this search also include closed issues seems like a low-effort way to achieve that.)

Reply:
Yes, that's what `state:open` is doing.
I did it this way strictly for ensuring that we are reporting it again if was re-introduced after fixing the initial created issue.
I was indeed considering that we will leave open the tickets for the issues that we are not going to fix. I know that this would cause some pollution but if we are tagging those accordingly it will be less of a burden.

Thanks for starting this conversation. This is definitely a decision that would be beneficial to have more input from the core contributors.

Reply:
I think that I would rather have the issue closed with a clear reason, instead of leaving them lingering.  It gives a weird view of the repo.

I think https://github.com/golang/go/issues/61211 would be ideal, but I don't see that happening any time soon.

Reply:
Removed the filtering on the state in 455033b8d10f0805dff9335c0ff931fd5beda3c6.